rxandroid/src/main/java/rx/android/lifecycle/OnSubscribeActivityLifecycleCallbacks.java

```diff
@@ -0,0 +1,156 @@
+package rx.android.lifecycle;
+
+import android.app.Activity;
+import android.app.Application;
+import android.os.Bundle;
+
+import rx.Observable;
+import rx.Subscriber;
+import rx.functions.Action1;
+import rx.functions.Func1;
+
+/**
+ * This instance lives within the scope of an {@code Activity} in order to forward Lifecycle events
+ * to {@link rx.subjects.BehaviorSubject}
+ * <p/>
+ * Thread safe, since it's <i>Thread-confined</i> to the main Thread.
+ */
+class OnSubscribeActivityLifecycleCallbacks implements Application.ActivityLifecycleCallbacks, Observable.OnSubscribe<LifecycleEvent> {
+    private final Activity activityToMonitor;
+    private Subscriber<? super LifecycleEvent> subscriber;
+
+    public OnSubscribeActivityLifecycleCallbacks(Activity instance) {
+        activityToMonitor = instance;
+    }
+
+    @Override
+    public void call(Subscriber<? super LifecycleEvent> subscriber) {
+        this.subscriber = subscriber;
+    }
+
+    @Override
+    public void onActivityCreated(final Activity activity, Bundle bundle) {
+        Observable
+                .just(activity)
+                .filter(sameInstance)
+                .subscribe(sendCreateEvent);
```

Comment:
Using an `Observable` in all these callbacks is wasteful in both allocation and performance. I don't see any advantage over
```java
if (activity == activityToMonitor) {
  subscriber.onNext(LifecycleEvent.CREATE);
}
```
which is allocation-free.